tests/integration/worker/test_submission_worker.py

```diff
@@ -0,0 +1,394 @@
+            mock_logger.assert_called_with("Exception raised while creating Python module for challenge_id: {}".format(self.challenge.id))
+
+
+@mock.patch("scripts.workers.submission_worker.SubmissionSerializer.data", "")
+@mock.patch("scripts.workers.submission_worker.SUBMISSION_DATA_DIR", "mocked/dir/submission_{submission_id}")
+@mock.patch("scripts.workers.submission_worker.PHASE_ANNOTATION_FILE_PATH", "mocked/dir/challenge_data/challenge_{challenge_id}/phase_data/phase_{phase_id}/test_annotation_file.txt")
+@mock.patch("scripts.workers.submission_worker.ContentFile")
+@mock.patch("scripts.workers.submission_worker.open")
+@mock.patch("scripts.workers.submission_worker.timezone")
+@mock.patch("scripts.workers.submission_worker.shutil")
+@mock.patch("scripts.workers.submission_worker.LeaderboardData.objects.bulk_create")
+@mock.patch("scripts.workers.submission_worker.create_dir")
+@mock.patch("scripts.workers.submission_worker.EVALUATION_SCRIPTS")
+@mock.patch("scripts.workers.submission_worker.PHASE_ANNOTATION_FILE_NAME_MAP")
+class RunSubmissionTestClass(BaseTestClass):
+    def setUp(self):
+        super(RunSubmissionTestClass, self).setUp()
+        self.leaderboard_schema = {
+            "labels": ["score", "test-score"],
+            "default_order_by": "score",
+        }
+        self.leaderboard = Leaderboard.objects.create(
+            schema=self.leaderboard_schema
+        )
+        self.dataset_split = DatasetSplit.objects.create(
+            name="Split 1", codename="split1"
+        )
+        self.challenge_phase_split = ChallengePhaseSplit.objects.create(
+            challenge_phase=self.challenge_phase,
+            dataset_split=self.dataset_split,
+            leaderboard=self.leaderboard,
+            visibility=ChallengePhaseSplit.PUBLIC,
+        )
+        self.leaderboard_data = LeaderboardData.objects.create(
+            challenge_phase_split=self.challenge_phase_split,
+            submission=self.submission,
+            leaderboard=self.leaderboard,
+            result={"metric1": 10},
+        )
+
+    def test_run_submission_when_result_key_is_not_there_in_output(self, mock_map, mock_script_dict,
+                                                                   mock_createdir, mock_lb,
+                                                                   mock_shutil, mock_timezone,
+                                                                   mock_open, mock_cf):
+        challenge_id = self.challenge.id
+        phase_id = self.challenge_phase.id
+        user_annotation_file_path = "tests/integration/worker/data/user_annotation.txt"
+        temp_run_dir = "mocked/dir/submission_{}/run".format(self.submission.id)
+
+        mock_map[challenge_id] = mock.Mock()
+        mock_map.get(challenge_id).get.return_value = "test_annotation_file.txt"
+        mock_script_dict[challenge_id] = mock.Mock()
+        mock_script_dict[challenge_id].evaluate.return_value = {"split1": {"metric1": 10}}
```

Comment:
Please don't hardcode this value. Use the value from the `leaderboard_data` created above, so that if we change that value then it should automatically get updated.